src/core/identifiers/identifiers.ts

```diff
@@ -24,6 +27,17 @@ export const validateDSNPAnnouncementId = (id: string): id is DSNPAnnouncementId
  */
 export type DSNPUserId = string;
 
+/**
+ * isDSNPUserId validates a given object as a DSNPUserId.
+ *
+ * @param id - The object to validate
+ * @returns True of false depending on whether the string is a valid DSNPUserId
+ */
+export const isDSNPUserId = (id: unknown): id is DSNPUserId => {
+  if (!isString(id)) return false;
+  return id.match(/^dsnp:\/\/[0-9A-F]{16}$/i) !== null;
```

Comment:
I'm working on this part of the spec, and a DSNP User Id right now does not have the `dsnp://` prefix

This is so that `fromId` and `objectId` don't need to have that prefix, but can still be referred and validated as DSNP User Ids.

Reply:
I thought the plan was to use binary in the parquet files and contracts and to immediately translate it to `dsnp://` hex strings everywhere in the SDK for user friendliness. Did that plan change?

Reply:
Maybe? Let's discuss with the team today.